packages/web-config-server/src/apiV1/utils/getMatchedNumeratorsAndDenominators.js

```diff
@@ -23,11 +24,32 @@ export const getMatchedNumeratorsAndDenominators = (
     }
   };
 
+  const sumValueByPeriod = results => {
+    const periodValues = {};
+    Object.keys(results).forEach(resultKey => {
+      const resultPeriod = results[resultKey].period;
+      if (!periodValues[resultPeriod]) {
+        periodValues[resultPeriod] = results[resultKey].value;
+      } else {
+        periodValues[resultPeriod] += results[resultKey].value;
+      }
+    });
+    return periodValues;
```

Comment:
I was really hoping we'd be able to avoid having this logic here, but I understand that it's a bit tricky.

I think ideally what we'd be doing here is:
1. Specify a different aggregation type for just the denominator that adds up the values within the country (`SUM_PER_PERIOD_PER_ORG_GROUP with { dataSourceEntityType: 'facility', aggregationEntityType: 'country' }`)
2. Pass in the `denominator.organisationUnitCode` into the `getMatchedNumeratorsAndDenominators()` and have it correctly match all numerators to that organisationUnitCode.

This way we can leave the `sumValuePerPeriod` logic to the aggregator.

However, I can see in this databuilder that the aggregation types are hardcoded unfortunately... It would be possible to make them overridable via config but that is a tad ugly. There are only 4 dashboards that use this databuilder, so it's feasible that we could change them to specify their aggregation type for numerator/denominator (which would allow us to just have 1 databuilder instead of two).

What are your thoughts on this? Happy to be pushed back on if you feel for practical reasons it's exceeding the scope of this bug fix

Reply:
I think I have to try it out before i have a strong opinion. Seems reasonable option to reduce the databuilder. Thanks for the suggestion. If it works out to be too obtuse then I consider pushing back :)